src/main.rs

```diff
@@ -748,46 +748,14 @@ fn get_setup() -> Setup {
         })
         .unwrap_or_default();
 
-    #[cfg(any(
-        feature = "alsa-backend",
-        feature = "rodio-backend",
-        feature = "portaudio-backend"
-    ))]
     let device = opt_str(DEVICE);
-
-    #[cfg(any(
-        feature = "alsa-backend",
-        feature = "rodio-backend",
-        feature = "portaudio-backend"
-    ))]
     if let Some(ref value) = device {
         if value == "?" {
             backend(device, format);
             exit(0);
-        } else if value.is_empty() {
-            empty_string_error_msg(DEVICE, DEVICE_SHORT);
```

Comment:
You still need that. It guards against empty strings as a device. `--device ""` is a valid argument as far as getopts is concerned but will fail.

Reply:
Good catch, I'll put that back in.